core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -795,6 +796,19 @@ public PlanOptimizers(
                         .add(new InlineProjections(plannerContext, typeAnalyzer))
                         .build()));
 
+        builder.add(new IterativeOptimizer(
+                plannerContext,
+                ruleStats,
+                statsCalculator,
+                costCalculator,
+                ImmutableSet.of(
+                        // Must run before AddExchanges
+                        new PushMergeWriterUpdateIntoConnector(metadata),
```

Comment:
Why not to add this rule to the below optimizer?

Reply:
Refactored